edb/schema/objects.py

```diff
@@ -1514,7 +1598,7 @@ def delta_sets(
             # and `old_schema`.
             comp = x.compare(
                 y,
-                our_schema=new_schema,
+                our_schema=new_schema,  # type: ignore
```

Comment:
add `assert new_schema is not None`?

Reply:
No, we can't assert it there, it's correlated with the collection being empty or not.  That whole function is a bit of a mess, we'll probably need to find a way to refactor it to avoid all this.